### PR TITLE
Fix AWS credentials

### LINKED ISSUE #110: V4.0.0 complains about S3 permissions
Version 4.0.0 of the plugin complains about S3 permission issues, whereas 3.2.0 does not (and works). To see if it was a missing permission I tried granting `s3*` to the bucket (via the AWS role of the box) but that didn't fix it. Reverted to 3.2.0 and S3 uploads worked again.

- Version: 4.0.0
- Operating System: ubuntu 16.04

Log entry:

```
{:timestamp=>"2017-01-03T00:27:44.601000+0000", :message=>"Pipeline aborted due to error", :exception=>"LogStash::ConfigurationError", :error=>"Logstash must have the privileges to write to root bucket `[snip]`, check you credentials or your permissions.", :backtrace=>["/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-s3-4.0.0/lib/logstash/outputs/s3.rb:187:in `register'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/output_delegator.rb:75:in `register'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/pipeline.rb:181:in `start_workers'", "org/jruby/RubyArray.java:1613:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/pipeline.rb:181:in `start_workers'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/pipeline.rb:136:in `run'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/agent.rb:491:in `start_pipeline'"], :level=>:error}
```

Config:

```
output {
     s3 {
         codec => "json_lines"
         bucket => "[snip]"
         size_file => 512000
         time_file => 5
         encoding => "gzip"
     }
}
```

### PR DESCRIPTION
**Pull request by joshuaspence**
Fixes #110. This plugin is not currently passing the correct credentials to the AWS SDK. Instead of trying to pass in `:credentials` explicitly, we should just trust that `LogStash::PluginMixins::AwsConfig::V2#aws_options_hash` does what it is expected. Currently, the plugin is failing in one of two different ways, when using instance profile credentials.

If `validate_credentials_on_root_bucket => true`, the following error is thrown during plugin registration:

```
{"level":"ERROR","loggerName":"logstash.agent","timeMillis":1483509033892,"thread":"[main]-pipeline-manager","logEvent":{"message":"Pipeline aborted due to error","exception":{"metaClass":{"metaClass":{"metaClass":{"exception":"Logstash must have the privileges to write to root bucket `REDACTED`, check you credentials or your permissions.","backtrace":["/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-s3-4.0.1/lib/logstash/outputs/s3.rb:187:in `register'","/usr/share/logstash/logstash-core/lib/logstash/output_delegator_strategies/shared.rb:8:in `register'","/usr/share/logstash/logstash-core/lib/logstash/output_delegator.rb:37:in `register'","/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:229:in `start_workers'","org/jruby/RubyArray.java:1613:in `each'","/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:229:in `start_workers'","/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:183:in `run'","/usr/share/logstash/logstash-core/lib/logstash/agent.rb:292:in `start_pipeline'"]}}}}}}
```

If `validate_credentials_on_root_bucket => false`, the following error is thrown from the uploader threads:

```
{"level":"ERROR","loggerName":"logstash.outputs.s3","timeMillis":1483506824991,"thread":"S3 output uploader, file: /tmp/logstash/b1fe230c-3df6-40ce-8293-6cc8133ba31c/ap-southeast-2/dt=2017-01-04/ls.s3.f4fb1158-e180-4fec-b355-bafc6f388563.2017-01-03T23.22.part3.txt","logEvent":{"message":"Uploading failed, retrying","exception":{"metaClass":{"metaClass":{"metaClass":{"exception":"unable to sign request without credentials set","path":"/tmp/logstash/b1fe230c-3df6-40ce-8293-6cc8133ba31c/ap-southeast-2/dt=2017-01-04/ls.s3.f4fb1158-e180-4fec-b355-bafc6f388563.2017-01-03T23.22.part3.txt","backtrace":["/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/request_signer.rb:100:in `require_credentials'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_request_signer.rb:14:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/xml/error_handler.rb:8:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_request_signer.rb:65:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_redirects.rb:15:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/retry_errors.rb:87:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_accelerate.rb:42:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_md5s.rb:31:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_expect_100_continue.rb:21:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_bucket_name_restrictions.rb:12:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_bucket_dns.rb:31:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/rest/handler.rb:7:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/user_agent.rb:12:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/seahorse/client/plugins/endpoint.rb:41:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/param_validator.rb:21:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/seahorse/client/plugins/raise_response_errors.rb:14:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_sse_cpk.rb:19:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/s3_accelerate.rb:33:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/seahorse/client/plugins/response_target.rb:21:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/seahorse/client/request.rb:70:in `send_request'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-core-2.3.22/lib/seahorse/client/base.rb:207:in `create_multipart_upload'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-resources-2.3.22/lib/aws-sdk-resources/services/s3/multipart_file_uploader.rb:52:in `initiate_upload'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-resources-2.3.22/lib/aws-sdk-resources/services/s3/multipart_file_uploader.rb:43:in `upload'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-resources-2.3.22/lib/aws-sdk-resources/services/s3/file_uploader.rb:32:in `upload'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/aws-sdk-resources-2.3.22/lib/aws-sdk-resources/services/s3/object.rb:251:in `upload_file'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-s3-4.0.1/lib/logstash/outputs/s3/uploader.rb:38:in `upload'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-s3-4.0.1/lib/logstash/outputs/s3/uploader.rb:29:in `upload_async'","org/jruby/RubyProc.java:281:in `call'","/usr/share/logstash/vendor/bundle/jruby/1.9/gems/concurrent-ruby-1.0.0-java/lib/concurrent/executor/java_executor_service.rb:94:in `run'","Concurrent$$JavaExecutorService$$Job_1168525423.gen:13:in `run'"]}}}}}}
```

### CODE REVIEW

**Comment by joshuaspence** [comment]
Hmm, the integration tests are failing on `master` when I run them locally.

**Comment by ph** [comment]
@joshuaspence they are passing for me, do you have a seed and which tests is failling for you? 
I am testing your PR.

File: spec/outputs/s3/file_repository_spec.rb [hunk 1]
```diff
@@ -105,9 +105,6 @@
 
     @file_repository.get_file("another-prefix") { |file| file.write("hello") }
     expect(@file_repository.size).to eq(2)
-    @file_repository.keys.each do |k|
-      puts k
-    end
```
Comment (ph) [inline spec/outputs/s3/file_repository_spec.rb]: good catch

**Comment by ph** [comment]
LGTM, I am +1 to merge this like this, for the failling integration test I think we can create another issue?

**Comment by ph** [comment]
I've ran the integration tests successfully.

```
Randomized with seed 30830

File Time rotation with stale write
  create one file
  Persists all events

Gzip Size rotation
  Rotates the files based on size
  Persists all events

Restore from crash
  uploads the file to the bucket

Upload current file on shutdown
  creates a specific quantity of files
  Persists all events

Upload current file on shutdown
  Persists all events

Dynamic Prefix
  with dates
    creates dated path
  With field string
    creates specific keys
    creates a specific quantity of files
    Persists all events
  with unsupported char
    convert them to underscore

File Time rotation with constant write
  Persists all events
  creates multiples files

Size rotation
  Persists all events
  creates a specific quantity of files
  Rotates the files based on size

Gzip File Time rotation with constant write
	  Persists all events
  creates multiples files

Finished in 4 minutes 58.8 seconds (files took 11.53 seconds to load)
20 examples, 0 failures

Randomized with seed 30830
```

**Pull request closed without merging**